src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,150 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import {getPreferredCurrency} from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+    /* Onyx Props */
+
+    // Url currently in view
+    currentURL: PropTypes.string,
+};
+
+// Determines type of step to display within Modal, value provides the title for that page.
+const Steps = {
+    IOUAmount: 'Amount',
+    IOUParticipants: 'Participants',
+    IOUConfirm: 'Confirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+
+        this.state = {
+            steps: [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm],
+            currentStepIndex: 0,
+            currentStepTitle: Steps.IOUAmount,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Navigate to the next IOU step if possible
+     */
+    navigateToPreviousStep() {
+        if (this.state.currentStepIndex <= 0) {
+            return;
+        }
+        this.setState(prevState => ({
+            currentStepIndex: prevState.currentStepIndex - 1,
+            currentStepTitle: prevState.steps[prevState.currentStepIndex - 1] || '',
+        }));
+    }
+
+    /**
+     * Navigate to the previous IOU step if possible
+     */
+    navigateToNextStep() {
+        if (this.state.currentStepIndex >= this.state.steps.length - 1) {
+            return;
+        }
+        this.setState(prevState => ({
+            currentStepIndex: prevState.currentStepIndex + 1,
+            currentStepTitle: prevState.steps[prevState.currentStepIndex + 1] || '',
+        }));
+    }
```

Comment:
I can't quite remember, but does it not make sense to use routes for the steps? Seems like it would be a little clearer.

Reply:
Addressed in the open thread, but this is on the way in later IOU issues